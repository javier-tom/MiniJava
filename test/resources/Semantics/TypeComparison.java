class Main {
    public static void main(String[] args) {
        System.out.println((new TypeCompare().run(10, 15)));
    }
}
class TypeCompare {
    int a;
    int b;
    int c;
    int[] arr;
    boolean flag;

    public int run(int a, int b) {
        a = 8; // precedence, should not change global a
        c = b; // should be parameter value
        arr = a + c; // not type compatible
        flag = 1; // not type compatible
        c = this.secondRun(a, flag); // parameter list not same type
        c = a + b;
        return flag; // error, return type does not match
    }

    public int secondRun(int x, int y) {
        a = x;
        System.out.println(a); // 8
        System.out.println(flag);
        arr = new int[flag]; // not type compatible
        return c;
    }
}