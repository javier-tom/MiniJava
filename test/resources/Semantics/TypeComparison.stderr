Error on line 16: cannot assign int to a int[]
Error on line 17: cannot assign int to a boolean
Error on line 18: expected type int in parameter 2 of the call to secondRun. Got type boolean instead.
Error on line 20: expected type int. Got type boolean
Error on line 26: expected type int. Got type boolean
Error on line 27: expected type int. Got type boolean
